lib/api_remote.ml

```diff
@@ -188,23 +182,24 @@ module Slack : Api.Slack = struct
     match url with
     | None ->
       Lwt.return
-      @@ build_error
-      @@ sprintf "no token or webhook configured to notify channel %s" (Slack_channel.Any.project msg.channel)
+      @@ Error
+           (build_error
+           @@ sprintf "no token or webhook configured to notify channel %s" (Slack_channel.Any.project msg.channel))
     | Some url ->
       let data = Slack_j.string_of_post_message_req msg in
       let body = `Raw ("application/json", data) in
       log#info "data: %s" data;
       if webhook_mode then begin
-        match%lwt http_request ~body ~headers `POST url with
-        | Ok _res ->
-          (* Webhooks reply only 200 `ok`. We can't generate anything useful for notification success handlers *)
-          Lwt.return @@ Ok None
-        | Error e -> Lwt.return @@ build_error (query_error_msg url e)
+        let* _res =
+          http_request ~body ~headers `POST url |> Lwt_result.map_error (fun e -> build_error (query_error_msg url e))
+        in
+        Lwt.return @@ Ok None
       end
       else begin
-        match%lwt slack_api_request ~body ~headers `POST url Slack_j.read_post_message_res with
-        | Ok res -> Lwt.return @@ Ok (Some res)
-        | Error e -> Lwt.return @@ build_error e
+        let* res =
+          slack_api_request ~body ~headers `POST url Slack_j.read_post_message_res |> Lwt_result.map_error build_error
+        in
+        Lwt.return @@ Ok (Some res)
```

Comment:
result of option type is not correct. I've mentioned this  before.

Either Ok of 'a, or Error of something. You can see by the code that there will never be an `Ok None`

Reply:
This is out of scope of this PR, this was present before.

I agree it would be nice to remove it though.

Reply:
Add it to your backlog, if  you can, otherwise i can do it. let me know